README.md
# storybuilder
[![Build Status](https://travis-ci.org/My-Novel-Management/storybuilder.svg?branch=master)](https://travis-ci.org/My-Novel-Management/storybuilder)
[![Coverage Status](https://coveralls.io/repos/github/My-Novel-Management/storybuilder/badge.svg)](https://coveralls.io/github/My-Novel-Management/storybuilder)

The story development tool by python

---
(C)2019 N.T.Works


CHANGELOG.md
# ChangeLog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.6.2] - 2020-10-02
### Changed
- World: change command shorter

## [0.6.1] - 2020-08-20
### Fixed
- Mecab arguments error

## [0.6.0] - 2020-07-31

## [0.5.9-24] - 2020-07-27
### Added
- StoryConfig: taginfos, chars and papers
- Outputter: datetime print on last
- Runner: output chars on console
### Changed
- Counter: is_contain_material
- Writer: into world.py
### Fixed
- asset: basic rubi

## [0.5.9-23] - 2020-07-20
### Added
- PlotUpdater
### Changed
- TagReplacer: config data replaced

## [0.5.9-22] - 2020-07-13
### Added
- Asset: common_rubi
### Fixed
- TagReplacer: tags order
- Compiler: then bug (description head nothing)

## [0.5.9-21] - 2020-07-10
### Added
- StoryConfig: caution, sites info
- StoryCommand: motif, foreshadow, payoff
- StoryConfig: copy, theme and more
- SceneInfoUpdater
### Changed
- Compiler: conv to plot (re-format)
### Fixed
- PercentAnalyzer: zero divided

## [0.5.9-20] - 2020-07-06
### Added
- Analyzer: percent analyze
- Analyzer: katakana and hiragana percent
- Analyzer: frequency analyze
### Fixed
- Analyzer: description characters count

## [0.5.9-19] - 2020-07-05
### Added
- Analyzer: word analyze
### Changed
- MyLogger: general reset level method
### Fixed
- ScriptConverter: scode end point using comma
- version string

## [0.5.9-18] - 2020-07-03
### Added
- SceneInfo
### Changed
- Material: not comment-like output in plot

## [0.5.9-17] - 2020-07-02
### Added
- DB: default person (who)
### Changed
- ScriptConverter: scode from comment
- last modified: automatically update from main.py

## [0.5.9-16] - 2020-07-02
### Added
- Story data output

## [0.5.9-15] - 2020-07-01
### Added
- Material: writer's note and character's note

## [0.5.9-14] - 2020-06-30
### Added
- SceneConverter

## [0.5.9-13] - 2020-06-30
### Added
- HeaderInfo: story version
- Runner: chapter part selection
### Fixed
- Compiler: text output with rubi-mode
- output plot-mode

## [0.5.9-12] - 2020-06-29
### Added
- Travis: test plot-mode and text-mode
- Compiler: text output
### Fixed
- SObject: omit priority

## [0.5.9-11] - 2020-06-25
### Added
- Counter: total count (lines and papers)
- Collecter class
- Compiler: contents title
### Changed
- Compiler: outline comment as a special format
### Fixed
- Chapter and Episode contain scode error
- Counter: Scmd get typo
- Counter: manupaper count
- StoryConfig: default outline set a string.

## [0.5.9-10] - 2020-06-23
### Added
- HeaderInfo: total chars
- Compiler: output plot
### Changed
- Logger option

## [0.5.9-9] - 2020-06-22
### Added
- CommentConverter

## [0.5.9-8] - 2020-06-22
### Added
- Asset: basic - times

## [0.5.9-7] - 2020-06-22
### Added
- Print StroyBuilder version at first

## [0.5.9-6] - 2020-06-21
### Fixed
- Compiler: with rubi

## [0.5.9-5] - 2020-06-21
### Changed
- Logger: default level setting

## [0.5.9-4] - 2020-06-21
### Changed
- StoryCmd: change_date and change_time new args

## [0.5.9-3] - 2020-06-21
### Fixed
- Typo: import file

## [0.5.9-2] - 2020-06-21
### Fixed
- DB: build from list

## [0.5.9-1] - 2020-06-19
### Fixed
- import error

## [0.5.9] - 2020-06-17
### Refined
- simplize all contents

## [0.5.1-4] - 2020-02-26
### Fixed
- rubi: kotatsu

## [0.5.1-3] - 2020-02-18
### Fixed
- rubi

## [0.5.1-2] - 2020-02-18
### Fixed
- format invalid breakline

## [0.5.1-1] - 2020-02-01
### Added
- episode char count in description

## [0.5.1] - 2020-02-01
### Added
- description output with scene char count

## [0.5.0] - 2020-02-01
### Added
- continued action

## [0.4.5-7] - 2020-01-31
### Changed
- person line format

## [0.4.5-6] - 2020-01-31
### Added
- person line output
### Changed
- stage and event line output dir

## [0.4.5-5] - 2020-01-31
### Fixed
- event point string longer

## [0.4.5-4] - 2020-01-31
### Fixed
- event line scene title

## [0.4.5-3] - 2020-01-31
### Changed
- event line format
### Fixed
- stage and area cooperation

## [0.4.5-2] - 2020-01-30
### Fixed
- Asset file names

## [0.4.5-1] - 2020-01-30
### Fixed
- Area not set bug

## [0.4.5] - 2020-01-30
### Added
- Area data class

## [0.4.4-5] - 2020-01-29
### Added
- elapsed day

## [0.4.4-4] - 2020-01-29
### Added
- Asset: acccessory
### Changed
- block output using list switch
### Fixed
- time delta in Stageline

## [0.4.4-3] - 2020-01-28
### Fixed
- duplicated maru in Conte

## [0.4.4-2] - 2020-01-28
### Added
- Time object new attr and methods: elapsed time

## [0.4.4-1] - 2020-01-28
### Fixed
- desc char count: except wear action strings

## [0.4.4] - 2020-01-28
### Changed
- object texture: single data from dict

## [0.4.3-33] - 2020-01-28
### Added
- scene objects output on Conte

## [0.4.3-32] - 2020-01-28
### Changed
- block info output using tags
### Fixed
- action object dupulicated when dividing actions

## [0.4.3-31] - 2020-01-28
### Changed
- stage line format

## [0.4.3-30] - 2020-01-27
### Fixed
- question and kakko bug: invalid inserting a space

## [0.4.3-29] - 2020-01-27
### Changed
- look action and subject display format on conte

## [0.4.3-28] - 2020-01-27
### Fixed
- combine to description bug: divided action

## [0.4.3-27] - 2020-01-27
### Changed
- stage line output format: time delta arrow

## [0.4.3-26] - 2020-01-27
### Added
- info volume output

## [0.4.3-25] - 2020-01-26
### Added
- event line output
### Changed
- stage line format

## [0.4.3-24] - 2020-01-26
### Added
- meta data: event

## [0.4.3-23] - 2020-01-26
### Fixed
- info tag convert

## [0.4.3-22] - 2020-01-25
### Added
- block title display on conte

## [0.4.3-21] - 2020-01-25
### Fixed
- part output end number

## [0.4.3-20] - 2020-01-24
### Fixed
- part output number

## [0.4.3-19] - 2020-01-24
### Added
- action per person output

## [0.4.3-18] - 2020-01-23
### Fixed

assets/basic.py
# -*- coding: utf-8 -*-
'''
Asset: basic data
=================
'''

ASSET = {
        "PERSONS":(
        # (tag / name / full / age (birth) / sex / job / call / info)
        ('man', '男', '', 30,(1,1), 'male', '会社員'),
        ('woman', '女', '', 30,(1,1), 'female', '会社員'),
        ),
        "STAGES":(
        # (tag / name / parent / (geometry) / info)
        # * basic area
        ('Japan', '日本', '', (135.00,35.00)),
        ('Tokyo', '東京', '', (139.67,35.76)),
        ),
        "DAYS":(
        # (tag / name / month / day / year)
        ('current', '現在', 4,1, 2020),
        ),
        "TIMES":(
        # (tag / name / hour / minute))
        ("earlymorning", "早朝", 6, 0),
        ("morning", "朝", 8, 0),
        ("midmorning", "午前", 10, 0),
        ("branch", "昼前", 11, 0),
        ("noon", "正午", 12, 0),
        ("afternoon", "午後", 14, 0),
        ("afterschool", "放課後", 16, 0),
        ("evening", "夕方", 17, 0),
        ("night", "夜", 20, 0),
        ("midnight", "深夜", 23, 0),
        ("deepnight", "真夜中", 2, 0),
        ),
        "ITEMS":(
        # (tag / name / cate / info)
        ),
        "WORDS":(
        # (tag / name / cate / info)
        ),
        "RUBIS":(
        # (origin / rubi / exclusions / always)
        ),
        }
